### PR TITLE
Bump jenkins to 2.452.3 (update Stapler) and Bom to `3010.vec758b_8e7da_3` (jQuery3 v3.7.1-2 and support for 2.452.x)

### PR DESCRIPTION
**Pull request by shlomomdahan**
Needed for pending CSP fixes. 

Jenkins version `2.452.3` includes:
1. changes to make Stapler's <st:bind> CSP compatible

Bom version [2839.v003b_4d9d24fd](https://github.com/jenkinsci/bom/releases?q=jquery3&expanded=true):
1. Updates jQuery3 from `3.7.1-1` to -> `3.7.1-2` but doesnt support 2.452.x line

Bom version [3010.vec758b_8e7da_3](https://github.com/jenkinsci/bom/releases/tag/3010.vec758b_8e7da_3)
- adds support for `2.452.x`line

ref:

- https://github.com/jenkinsci/build-pipeline-plugin/pull/152

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -292,7 +293,7 @@
         </repository>
         <snapshotRepository>
              <id>maven.jenkins-ci.org</id>
-	     <url>https://repo.jenkins-ci.org/snapshots</url>
+             <url>https://repo.jenkins-ci.org/snapshots</url>
```
Comment (basil) [inline pom.xml:296]: Formatting change unrelated to the core diff?

File: pom.xml [hunk 2]
```diff
@@ -28,7 +28,7 @@
     <properties>
         <revision>2.0.3</revision>
         <changelist>-SNAPSHOT</changelist>
-        <jenkins.version>2.414.3</jenkins.version>
+        <jenkins.version>2.452.3</jenkins.version>
```
Comment (basil) [inline pom.xml:31]: BOM version on line 55 needs to be updated to match.

File: pom.xml [hunk 3]
```diff
@@ -85,6 +85,7 @@
         <dependency>
             <groupId>io.jenkins.plugins</groupId>
             <artifactId>jquery3-api</artifactId>
+            <version>3.7.1-2</version>
```
Comment (basil) [inline pom.xml:88]: Rather update the BOM version on line 55 to one that contains this release.

**Comment by shlomomdahan** [comment]
@basil 
I have made the changes to update the Bom instead of explicitly defining jQuery version number.

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Pull request merged**